pkg/operation/botanist/clusteridentity/clusteridentity.go

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusteridentity
+
+import (
+	"context"
+	"fmt"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type shootClusterIdentity struct {
+	clusterIdentity *string
+	seedNamespace   string
+	namespace       string
+	name            string
+	uid             string
+
+	seedClient   client.Client
+	gardenClient client.Client
+
+	logger *logrus.Entry
+}
+
+func New(
+	clusterIdentity *string,
+	name, namespace, seedNamespace, uid string,
+	seedClient, gardenClient client.Client,
+	logger *logrus.Entry,
+) component.Deployer {
+	return &shootClusterIdentity{
+		clusterIdentity: clusterIdentity,
+		name:            name,
+		namespace:       namespace,
+		uid:             uid,
+		seedNamespace:   seedNamespace,
+		seedClient:      seedClient,
+		gardenClient:    gardenClient,
+		logger:          logger,
+	}
+}
+
+func (c *shootClusterIdentity) Deploy(ctx context.Context) error {
+	if c.clusterIdentity == nil {
+		clusterIdentity, err := c.determineShootClusterIdentity(ctx)
+		if err != nil {
+			return err
+		}
+
+		shoot := &gardencorev1beta1.Shoot{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      c.name,
+				Namespace: c.namespace,
+			},
+		}
+
+		c.clusterIdentity = &clusterIdentity
+		patch := []byte(fmt.Sprintf(`{"status":{"clusterIdentity": "%s"}`, clusterIdentity))
+		if err := c.gardenClient.Status().Patch(ctx, shoot, client.RawPatch(types.StrategicMergePatchType, patch)); err != nil {
```

Comment:
Any specific reason why you use `PATCH` here?

Reply:
To avoid passing the Shoot as dependency, conflict during update or GET for the latest Shoot object.

Reply:
Hm, ok, fine. OTOH, you probably could put this whole block into an `if` block that is only executed when the `.status.clusterIdentity` of the shoot is not set yet. Otherwise, nothing needs to be done and you can exit early?